File: agent/verticals/automotive.py
"""Automotive vertical definition."""

from agent.verticals.base import VerticalDefinition, tabs

VERTICAL = VerticalDefinition(
    key="automotive",
    label="Automotive",
    risk_level="medium",
    entity_label_singular="vehicle",
    entity_label_plural="vehicles",
    default_plan_label="Automotive plan",
    crm_tabs=tabs(
        ("overview", "Overview"),
        ("readiness", "Readiness"),
        ("catalog", "Vehicles"),
        ("calculators", "Finance"),
        ("leads", "Test Drives"),
        ("activity", "Conversations"),
        ("prompt", "Prompt"),
        ("controls", "Controls"),
    ),
    entity_types=("vehicle_listing", "vehicle_model", "trim", "dealer", "finance_offer", "test_drive_slot"),
    readiness_checks=("vehicles", "specs", "dealer_contact", "test_drive", "finance", "freshness"),
    action_types=("SHOW_ENTITIES", "COMPARE_ENTITIES", "REQUEST_TEST_DRIVE", "RUN_CALCULATOR", "CAPTURE_LEAD"),
)

File: crm/src/verticals/definitions/automotive.ts
import { tab } from '../shared';
import type { CrmVerticalDefinition } from '../types';

export const automotiveVertical: CrmVerticalDefinition = {
  key: 'automotive',
  label: 'Automotive',
  riskLevel: 'medium',
  entityLabelSingular: 'vehicle',
  entityLabelPlural: 'vehicles',
  defaultPlanLabel: 'Automotive plan',
  clientTabs: [
    tab('overview', 'Overview'),
    tab('readiness', 'Readiness'),
    tab('catalog', 'Vehicles'),
    tab('calculators', 'Finance'),
    tab('leads', 'Test Drives'),
    tab('activity', 'Conversations'),
    tab('prompt', 'Prompt'),
    tab('controls', 'Controls'),
  ],
  entityTypes: ['vehicle_listing', 'vehicle_model', 'trim', 'dealer', 'finance_offer', 'test_drive_slot'],
  readinessChecks: ['vehicles', 'specs', 'dealer_contact', 'test_drive', 'finance', 'freshness'],
};
